File: main/python/insertion_sort.py
from random import shuffle

ARRAY_LENGTH = 100

def insertion_sort(array):
 
    for i in range(1, ARRAY_LENGTH):
 
        key = array[i]
 
        j = i-1
        while j >= 0 and key < array[j]:
                array[j + 1] = array[j]
                j -= 1
        array[j + 1] = key

array = []

for n in range(ARRAY_LENGTH):
    array.append(n+1)

shuffle(array)

insertion_sort(array)

File: main/c++/algorithms/insertion_sort.cpp
#include "sort.h"

void insertion_sort(int *array, int len) {
    for (int i = 1; i < len; i++) {
        int key = array[i];
        int j = i-1;
        while (j >= 0 and key < array[j]) {
            array[j + 1] = array[j];
            j -= 1;
        }
        array[j + 1] = key;
    }
}